src/github.com/matrix-org/dendrite/clientapi/routing/register.go

```diff
@@ -810,3 +562,32 @@ func RegisterAvailable(
 		},
 	}
 }
+
+// validateApplicationService checks if a provided application service token
+// corresponds to one that is registered. If so, then it checks if the desired
+// username is within that application service's namespace. As long as these
```

Comment:
It no longer checks the username. Please remember to update comments when you change code so that we so not end up with misleading comments.

Reply:
In fact, why have you factored out the username check to a different function? It seems to belong in this function.